lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -83,3 +83,107 @@ function( $settings ) {
 	},
 	100
 );
+
+
+if ( ! wp_is_block_theme() ) {
+	// Load the block styles inline for non-block themes.
+	add_filter( 'should_load_separate_core_block_assets', '__return_true', 1 );
+	// Remove the block styles from the footer.
+	remove_action( 'wp_footer', 'wp_maybe_inline_styles', 1 );
+}
+
+/**
+ * Print block styles inline for non-block (classic) themes.
+ * Improves performance for classic themes by only loading the styles that are needed
+ * for the blocks that are actually used on the page, instead of loading all block styles.
+ * At the same time, this reduces layout shifts by inlining these styles
+ * instead of printing them all in the footer.
+ *
+ * NOTE: When backported to Core,
+ * this should probably be merged with the `wp_maybe_inline_styles` function.
+ *
+ * This function is hooked on `render_block`.
+ *
+ * @param string $block_content The block content.
+ * @param array  $block         The full block, including name and attributes.
+ *
+ * @return string The block content.
+ */
+function gutenberg_inline_block_styles_on_classic_themes( $block_content, $block ) {
+	// Bail early if not a classic theme.
+	if ( wp_is_block_theme() ) {
```

Comment:
I think we also need to check `wp_should_load_separate_core_block_assets()` here. That decouples this logic from the concrete filter change above. For example, if somebody opts out via the filter, or if maybe the opt-in shouldn't be by default, things will still work correctly by that.
```suggestion
	if ( wp_is_block_theme() || ! wp_should_load_separate_core_block_assets() ) {
```